src/main/resources/templates/login.html

```diff
@@ -2,48 +2,180 @@
+            <ul class="navbar-nav ml-auto">
+                <li class="nav-item">
+                    <a class="nav-link" href="/"><i class="fas fa-home mr-1"></i> Home</a>
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" href="/browse"><i class="fas fa-compass mr-1"></i> Browse</a>
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" href="/login"><i class="fas fa-user mr-1"></i> Profile</a>
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" href="/settings"><i class="fas fa-cog mr-1"></i> Settings</a>
+                </li>
+            </ul>
+        </div>
+    </div>
+</nav>
+
+
+<div class="container login-container my-5">
+    <div class="text-center mb-4">
+        <h2 class="font-weight-bold">Welcome Back!</h2>
+        <p class="text-muted">Please login to your account</p>
+    </div>
+
+    <div class="login-card">
+        <form th:action="@{/signin}" th:object="${signin}" method="post">
+            <!-- Username Field -->
+            <div class="form-group">
+                <label for="username" class="font-weight-bold">Username</label>
+                <div class="input-group">
+                    <div class="input-group-prepend">
+                            <span class="input-group-text bg-white border-right-0">
+                                <i class="fas fa-user text-primary"></i>
+                            </span>
+                    </div>
+                    <input type="text"
+                           class="form-control border-left-0"
+                           id="username"
+                           th:field="*{username}"
+                           placeholder="Enter your username">
+                </div>
+                <span th:if="${#fields.hasErrors('username')}"
+                      class="text-danger small"
+                      th:errors="*{username}"></span>
+            </div>
+
+
+            <div class="form-group">
+                <label for="password" class="font-weight-bold">Password</label>
+                <div class="input-group">
+                    <div class="input-group-prepend">
+                            <span class="input-group-text bg-white border-right-0">
+                                <i class="fas fa-lock text-primary"></i>
+                            </span>
+                    </div>
+                    <input type="password"
+                           class="form-control border-left-0"
+                           id="password"
+                           th:field="*{password}"
+                           placeholder="Enter your password">
+                </div>
+                <span th:if="${#fields.hasErrors('password')}"
+                      class="text-danger small"
+                      th:errors="*{password}"></span>
+            </div>
```

Comment:
For this I used Thymeleaf form bindings to connect the login form to the signin object in the controller. The th:action attribute directs form submission to the /signin endpoint, and th:object binds the form fields to the signin class in the DTO. The th:field attributes map input fields to username and password properties in signin, while th:if and th:errors are used to display validation error messages. This was done to ensure dynamic form validation feedback and data binding to the backend.